src/openrct2/ride/ride.c

```diff
@@ -1090,9 +1090,12 @@ void ride_clear_for_construction(sint32 rideIndex)
     ride->lifecycle_flags &= ~(RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN);
     ride->window_invalidate_flags |= RIDE_INVALIDATE_RIDE_MAIN | RIDE_INVALIDATE_RIDE_LIST;
 
+    // Fixes another MP desync where it left the rating set to the previous run causing peeps to pick different rides.
+    invalidate_test_results(rideIndex);
```

Comment:
Just clearing for construction should _not_ invalidate the test results. It's very annoying to the user and prevents the 'double close and open' trick. If there is absolutely no other way, at least wrap this in an if that checks for multiplayer.

I'd also like to know how this would cause desyncs. Just clearing for construction does not affect the rating, only actually changing the ride will.